File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CreateExperimentDetails.java
package com.oracle.aidataplatform.dp.model;


/**
 * The data to create an experiment.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CreateExperimentDetails.Builder.class)

public final class CreateExperimentDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"name", "artifactLocation", "tags"})
    public CreateExperimentDetails(String name, String artifactLocation, java.util.List<ExperimentTag> tags) {
        super();
        this.name = name;
        this.artifactLocation = artifactLocation;
        this.tags = tags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Name of the experiment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("name")
        private String name;

                /**
         * Name of the experiment.
         * @param name the value to set
         * @return this builder
         **/
        

        public Builder name(String name) {
        this.name = name;
        return this;
        }
            /**
     * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("artifact_location")
        private String artifactLocation;

                /**
         * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
         * @param artifactLocation the value to set
         * @return this builder
         **/
        

        public Builder artifactLocation(String artifactLocation) {
        this.artifactLocation = artifactLocation;
        return this;
        }
            /**
     * List of tags set on the experiment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("tags")
        private java.util.List<ExperimentTag> tags;

                /**
         * List of tags set on the experiment.
         * @param tags the value to set
         * @return this builder
         **/
        

        public Builder tags(java.util.List<ExperimentTag> tags) {
        this.tags = tags;
        return this;
        }


        public CreateExperimentDetails build() {
            CreateExperimentDetails model = new CreateExperimentDetails(this.name
                    , this.artifactLocation
                    , this.tags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CreateExperimentDetails model) {
                this.name(model.getName());
    this.artifactLocation(model.getArtifactLocation());
    this.tags(model.getTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Name of the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * Name of the experiment.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("artifact_location")
    private final String artifactLocation;

        /**
     * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
     * @return the value
     **/
    
    public String getArtifactLocation() {
        return artifactLocation;
    }


        /**
     * List of tags set on the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tags")
    private final java.util.List<ExperimentTag> tags;

        /**
     * List of tags set on the experiment.
     * @return the value
     **/
    
    public java.util.List<ExperimentTag> getTags() {
        return tags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CreateExperimentDetails(");
        sb.append("name=").append(String.valueOf(this.name));
        sb.append(", artifactLocation=").append(String.valueOf(this.artifactLocation));
        sb.append(", tags=").append(String.valueOf(this.tags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CreateExperimentDetails)) {
            return false;
        }

        CreateExperimentDetails other = (CreateExperimentDetails) o;
        return java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.artifactLocation, other.artifactLocation) &&
            java.util.Objects.equals(this.tags, other.tags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.artifactLocation == null ? 43 : this.artifactLocation.hashCode());
        result = (result * PRIME) + (this.tags == null ? 43 : this.tags.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/create_experiment_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CreateExperimentDetails(object):
    """
    The data to create an experiment.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CreateExperimentDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param name:
            The value to assign to the name property of this CreateExperimentDetails.
        :type name: str

        :param artifact_location:
            The value to assign to the artifact_location property of this CreateExperimentDetails.
        :type artifact_location: str

        :param tags:
            The value to assign to the tags property of this CreateExperimentDetails.
        :type tags: list[oci.aidataplatform_dp.models.ExperimentTag]

        """
        self.swagger_types = {
            'name': 'str',
            'artifact_location': 'str',
            'tags': 'list[ExperimentTag]'
        }

        self.attribute_map = {
            'name': 'name',
            'artifact_location': 'artifact_location',
            'tags': 'tags'
        }

        self._name = None
        self._artifact_location = None
        self._tags = None

    @property
    def name(self):
        """
        **[Required]** Gets the name of this CreateExperimentDetails.
        Name of the experiment.


        :return: The name of this CreateExperimentDetails.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this CreateExperimentDetails.
        Name of the experiment.


        :param name: The name of this CreateExperimentDetails.
        :type: str
        """
        self._name = name

    @property
    def artifact_location(self):
        """
        Gets the artifact_location of this CreateExperimentDetails.
        Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.


        :return: The artifact_location of this CreateExperimentDetails.
        :rtype: str
        """
        return self._artifact_location

    @artifact_location.setter
    def artifact_location(self, artifact_location):
        """
        Sets the artifact_location of this CreateExperimentDetails.
        Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.


        :param artifact_location: The artifact_location of this CreateExperimentDetails.
        :type: str
        """
        self._artifact_location = artifact_location

    @property
    def tags(self):
        """
        Gets the tags of this CreateExperimentDetails.
        List of tags set on the experiment.


        :return: The tags of this CreateExperimentDetails.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentTag]
        """
        return self._tags

    @tags.setter
    def tags(self, tags):
        """
        Sets the tags of this CreateExperimentDetails.
        List of tags set on the experiment.


        :param tags: The tags of this CreateExperimentDetails.
        :type: list[oci.aidataplatform_dp.models.ExperimentTag]
        """
        self._tags = tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
